WeakAurasOptions/OptionsFrames/TextEditor.lua

```diff
@@ -77,22 +109,106 @@ local function ConstructTextEditor(frame)
   -- The indention lib overrides GetText, but for the line number
   -- display we ned the original, so save it here.
   local originalGetText = editor.editBox.GetText;
-  IndentationLib.enable(editor.editBox, colorScheme, 4);
 
   local cancel = CreateFrame("Button", nil, group.frame, "UIPanelButtonTemplate");
   cancel:SetScript("OnClick", function() group:CancelClose() end);
   cancel:SetPoint("BOTTOMRIGHT", -27, 13);
+  cancel:SetFrameLevel(cancel:GetFrameLevel() + 1)
   cancel:SetHeight(20);
   cancel:SetWidth(100);
   cancel:SetText(L["Cancel"]);
 
   local close = CreateFrame("Button", nil, group.frame, "UIPanelButtonTemplate");
   close:SetScript("OnClick", function() group:Close() end);
   close:SetPoint("RIGHT", cancel, "LEFT", -10, 0)
+  close:SetFrameLevel(close:GetFrameLevel() + 1)
   close:SetHeight(20);
   close:SetWidth(100);
   close:SetText(L["Done"]);
 
+  IndentationLib.enable(editor.editBox, get_scheme(WeakAurasSaved["editor_themes"]["selected"]), 4)
+  local is_settings_open = false
+  local settings_frame = CreateFrame("Button", "WASettingsButton", close, "UIPanelButtonTemplate")
+  settings_frame:SetPoint("RIGHT", close, "LEFT", -10, 0)
+  settings_frame:SetHeight(20)
+  settings_frame:SetWidth(100)
+  settings_frame:SetText("Settings")
+  settings_frame:EnableMouse(true)
+  settings_frame:RegisterForClicks("AnyDown")
+
+  local menu_frame = CreateFrame("Frame", "SettingsMenuFrame", settings_frame, "UIDropDownMenuTemplate")
+  menu_frame:SetPoint("CENTER", settings_frame, "Center")
+  menu_frame:Hide()
+
+  local function settings_menu()
+    local menu = {}
+    -- themes options
+    for k, v in pairs(WeakAurasSaved["editor_themes"]["themes"]) do
+      local item = {
+        text = k,
+        isNotRadio = false,
+        checked = (WeakAurasSaved["editor_themes"]["selected"] == k and true) or false,
+        func = function()
+          is_settings_open = false
+          WeakAurasSaved["editor_themes"]["selected"] = k
+          IndentationLib.enable(editor.editBox, get_scheme(k), 4)
+          editor.editBox:SetText(editor.editBox:GetText())
+        end
+      }
+      table.insert(menu, item)
+    end
+    -- bracket matching option
+    table.insert(menu, {
+      text = "Bracket Matching",
+      isNotRadio = true,
+      checked = WeakAurasSaved["bracket_matching"],
+      func = function()
+        is_settings_open = false
+        WeakAurasSaved["bracket_matching"] = not WeakAurasSaved["bracket_matching"]
+      end
+    })
+    return menu
+  end
+
+  settings_frame:SetScript("OnClick", function(self, button, down)
+    if button == "LeftButton" then
+      if is_settings_open then
+        ToggleDropDownMenu(1, nil, menu_frame, settings_frame, 0, 0, settings_menu(), nil, 0)
+        is_settings_open = false
+      else
+        EasyMenu(settings_menu(), menu_frame, settings_frame, 0, 0, "MENU")
```

Comment:
I think it should be possible to use ToggleDropDownMenu to open the menu. That would remove the need for the is_settings_open variable and simplify the code a bit.